packages/playwright-core/src/server/dispatchers/browserTypeDispatcher.ts

```diff
@@ -62,7 +64,9 @@ export class BrowserTypeDispatcher extends Dispatcher<BrowserType, channels.Brow
     controller.setLogName('browser');
     return await controller.run(async progress => {
       const paramsHeaders = Object.assign({ 'User-Agent': getUserAgent() }, params.headers || {});
-      const transport = await WebSocketTransport.connect(progress, params.wsEndpoint, paramsHeaders, true);
+      const wsEndpoint = await urlToWSEndpoint(progress, params.wsEndpoint);
```

Comment:
```suggestion
      const wsEndpoint = await urlToWSEndpoint(progress, params.wsEndpoint);
```
```suggestion
      const wsEndpoint = await urlToWSEndpoint(progress, params.wsEndpoint, paramsHeaders);
```

You want to make sure that the right headers gets passed there including user agent.

Reply:
These headers should not be passed to ws endpoint resolvement.